File: java/copilot_example.java
// This method works, but could be improved for readability and long-term maintenance.
// Your challenge: reduce technical debt without changing the output.

public class MessyExample {
    public static void main(String[] args) {
        String[] arr = {"apple", "banana", "cherry", "date"};
        String v = "a";
        for (int i=0;i<arr.length;i++) {
            if (arr[i].contains(v)) {
                System.out.println(arr[i]+i);
            }
        }
    }
}

File: go/copilot_example.go
// This function works, but could be improved for readability and long-term maintenance.
// Your challenge: reduce technical debt without changing the output.

package main

import "fmt"
import "strings"

func main() {
    arr := []string{"one", "two", "three", "four"}
    val := "o"
    var r []string
    for i := 0; i < len(arr); i++ {
        if strings.Contains(arr[i], val) {
            r = append(r, arr[i]+fmt.Sprintf("%d", i))
        }
    }
    fmt.Println("Result:")
    for _, v := range r {
        fmt.Println(v)
    }
}
